### ISSUE TITLE
Breaking links on save

### ISSUE THREAD

**Issue by bwmarrin** [fatih/vim-go#774]
### Actual behavior

When I edit a symlink or hardlink of a .go file and save it, the link is broken.

### Expected behavior

I expect the link to still be a link.

### Steps to reproduce:

```
touch test.go
ln -s test.go new.go
ls -l
total 1
lrwxr-xr-x  1 bruce  bruce  7 Mar 23 16:50 new.go -> test.go
-rw-r--r--  1 bruce  bruce  0 Mar 23 16:50 test.go
vim new.go
ls -l
total 5
-rw-r--r--  1 bruce  bruce  13 Mar 23 16:51 new.go
-rw-r--r--  1 bruce  bruce   0 Mar 23 16:50 test.go
```

### Configuration

* `vimrc` : https://github.com/bwmarrin/dotfiles
* vim version: 7.4.898
* vim-go version: 1.5
* go version: 1.5.3
* OS : FreeBSD 10

**Comment by fatih** [comment]
Hi @bwmarrin 

Thanks for the bug report. Can you please try this fix: https://github.com/fatih/vim-go/pull/783

Thanks

**Comment by bwmarrin** [comment]
Hey! Awesome :)

That seems to fix it for symbolic links, but it is still breaking for a hard link.

**Comment by fatih** [comment]
@bwmarrin seems like there is an option to follow links, checkout:

http://stackoverflow.com/questions/13563627/how-does-vim-save-files
http://vim.wikia.com/wiki/Editing_a_hard_link_to_a_file

There is no other way to me to follow hard links, I'm already using Vim's own function.

**Comment by bwmarrin** [comment]
I'll read though these and report back. I am traveling this week so it might be a couple days before I can test things. When I did my previous testing though, using the exact same vimrc when editing any other file (non .go file) vim did not break hard or symbolic links.  Not sure if that helps at all..

**Comment by bwmarrin** [comment]
Sorry for the delay but I remembered this today when I saw teh 1.6 post on reddit.  So I quickly updated to 1.6 and decided to give this a try again.  So, soft-links seem to not be altered and writing to a softlink will update the original file and not break the link.  Perfect.

Hard links however are still breaking.  I tested the exact same thing with non `.go` files like a `.mk` file and with anything other than a `.go` file vim isn't breaking the hardlink, the files remain linked.  So I do wonder where the difference is but there must be something different that it's doing specifically with `.go` files and vim-go.

I read the two documents you linked then read the vim help page which actually was a bit clearer for me.  My `backupcopy` setting is set to `auto` by default.  So I tried setting it to `yes` which is suppose to ensure that hardlinks are not broken.  But, the hardlink was still broken.

So, I fiddle some more and I set `let g:go_fmt_autosave = 0` then I don't have the problem anymore.  I also found this https://www.bountysource.com/issues/28660612-remote-editing-wipes-out-go-files-with-go_fmt_autosave-enabled while searching for possible solutions.

So it seems like it's at least related to the gofmt part.

This isn't a huge deal for me since I rarely work on hardlinks (I do use softlinks a lot so having that fixed is a big win, thank you very much).

Hope all that info helps somehow and sorry for the rant-ish length.
